lib/interferon.rb

```diff
@@ -210,6 +224,7 @@ def update_alerts(destinations, hosts, alerts, groups)
         threads = concurrency.times.map do
           t = Thread.new do
             while name = alerts_to_create.shift
+              break if @request_shutdown
```

Comment:
i would avoid creating threads in the first place, rather than creating threads which immediately exit.